### PR TITLE
Bitcoin bid interface

### LINKED ISSUE #10: Bitcoin-funded bid submission
TBD

### PR DESCRIPTION
**Pull request by pavel-kiselyov**
closes #10 

TODO:
- [x] Bitcoin bid interface & unit tests
- [x] Various fixes from PR #31 code review

### CODE REVIEW

File: contracts/DutchAuction.sol [hunk 1]
```diff
@@ -33,7 +34,7 @@ contract DutchAuction {
     event AuctionSetup();
     event AuctionStarted();
     event AuctionEnded(uint256 priceFinal, Endings ending);    
-    event BidAccepted(address indexed _address, uint256 price, uint256 transfer);
+    event BidAccepted(address indexed _address, uint256 price, uint256 transfer, bool isBitcoin);
```
Comment (ajhurliman) [inline contracts/DutchAuction.sol]: I didn't realize we could accept BTC... I'll need to look into this code more.

File: migrations/2_deploy_contracts.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 var DutchAuction = artifacts.require("DutchAuction");
 var ShopToken = artifacts.require("ShopToken");
 
-module.exports = function (deployer) {
+module.exports = function (deployer, network, accounts) {
   // Mint 1B tokens, transfer 100K for dutch auction
   const multiplier = 10 ** 18;
   const initialSupply = (10 ** 9) * multiplier;
```
Comment (ajhurliman) [inline migrations/2_deploy_contracts.js]: The largest number JS can handle is 2^53 - 1 (i.e. 9.007 * 10^15), so we might need to get something else to handle this number. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

Comment (pavel-kiselyov) [reply migrations/2_deploy_contracts.js]: Great, I will use BigNumber.js that is already integrated with tests and web3.js

File: contracts/DutchAuction.sol (hunk 1 above)
Comment (pavel-kiselyov) [reply contracts/DutchAuction.sol]: `placeBitcoinBid` will be called by a separate application called `coinflip`, which will listen to Bitcoin transaction to our wallets.

**Pull request merged**